pkg/kopia/repositoryserver/pkg/controller/repositoryserver.go

```diff
@@ -0,0 +1,186 @@
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "repository-server"})
+
+	controller := &Controller{
+		rsClientSet:   clientset,
+		rsLister:      rsInformer.Lister(),
+		workQueue:     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "repository-server"),
+		rsCacheSynced: rsInformer.Informer().HasSynced,
+		recorder:      recorder,
+	}
+
+	klog.Info("Setting up event handlers")
+	// Set up an event handler for when RepositoryServer resources change
+	rsInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    controller.handleAdd,
+			DeleteFunc: controller.handleDel,
+		},
+	)
+
+	return controller
+}
+
+func (c *Controller) handleAdd(obj interface{}) {
+	log.Println("handleAdd was called")
+	c.workQueue.Add(obj)
+}
+
+func (c *Controller) handleDel(obj interface{}) {
+	log.Println("handleDel was called")
+	c.workQueue.Done(obj)
+}
+
+func (c *Controller) Run(stopCh <-chan struct{}) error {
+	log.Println("Starting RepositoryServer Controller....")
+	if !cache.WaitForCacheSync(stopCh, c.rsCacheSynced) {
+		log.Println("Waiting for the cache to be synced....")
+	}
+
+	go wait.Until(c.worker, 1*time.Second, stopCh)
+
+	<-stopCh
+	return nil
+}
+
+func (c *Controller) worker() {
+	for c.processItem() {
+	}
+}
+
+func (c *Controller) processItem() bool {
+	item, shutdown := c.workQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	defer c.workQueue.Forget(item)
+
+	key, err := cache.MetaNamespaceKeyFunc(item)
+	if err != nil {
+		log.Printf("Error %s calling Namespace key func on cache for item", err.Error())
+		return false
+	}
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		log.Printf("Splitting key into namespace and name, error %s\n", err.Error())
+		return false
+	}
+
+	err = c.logOutputAfterCreateCR(ns, name)
+	if err != nil {
+		log.Println("Error while listing CR: ", err.Error())
+		return false
+	}
+	return true
+}
+
+func (c *Controller) logOutputAfterCreateCR(ns, name string) error {
+	reposerver, err := c.rsLister.RepositoryServers(ns).Get(name)
+	if err != nil {
+		log.Printf("Error in listing RepositoryServer resource: %s", err.Error())
+		return err
+	}
+
+	log.Printf("Successfully created RepositoryServer CR named %s", reposerver.Name)
+	return nil
+}
+
+func (c *Controller) startRepositoryServerPod() {
+
+}
+
+func (c *Controller) createService() {
+
+}
+
+func (c *Controller) createNetworkPolicy() {
+
+}
+
+func (c *Controller) addTLSCertConfigurationInPodOverride() {
+
+}
+
+func (c *Controller) createRepoServerPod() {
+
+}
+
+func (c *Controller) waitForPodReady() {
+
+}
+
+func (c *Controller) connectToRepository() {
+
+}
+
+func (c *Controller) startRepoProxyServer() {
+
+}
+
+func (c *Controller) waitForServerToStart() {
+
+}
+
+func (c *Controller) addClientUsersToServer() {
+
+}
+
+func (c *Controller) refreshServer() {
+
+}
```

Comment:
Might wanna consider moving all these methods to a `RepositoryServer` struct for better abstraction design.

Reply:
Noted @ihcsim ✅ This is WIP at the moment

Reply:
Moved the changes suggested in this comment to PR https://github.com/kanisterio/kanister/pull/1653

Reply:
@kale-amruta is this conversation addressed? If yes, can we please resolve this?

Reply:
yes Vivek